### PR TITLE
feat(rwx): add rwx nfsOptions to robot test scaffolding.

### PR DESCRIPTION
**Pull request by james-munson**
#### Which issue(s) this PR fixes:

Issue # https://github.com/longhorn/longhorn/issues/7639

#### What this PR does / why we need it:

This is a draft for adding a new storage class and rwx volume with overridden mount options to an `e2e` test.

#### Special notes for your reviewer:

Untested - hope to test locally or when the pipeline can be triggered with a private build of longhorn-tests.  Just checking to see that this is the way it would be done.

### CODE REVIEW

**Comment by yangchiu** [comment]
@james-munson That's great to see team members contribute to negative testing code. Many thanks for the active involvement! It's greatly appreciated!

Currently PRs for negative testing are blocked by the refactoring PR. I'll try to merge this once the refactoring PR is done.

> Untested - hope to test locally.

I'll try to test locally when I'm available.

File: e2e/libs/workload/workload.py
Comment (yangchiu) [inline e2e/libs/workload/workload.py:27]: Tested based on https://github.com/longhorn/longhorn-tests/pull/1805.

`storageclass-nfs-options.yaml` should be `storageclass_nfs_options.yaml`.
Others LGTM.

**Review by yangchiu — CHANGES_REQUESTED** [review]

**Review by khushboo-rancher — APPROVED** [review]

File: e2e/libs/workload/workload.py
Comment (james-munson) [reply e2e/libs/workload/workload.py:27]: I'll make that change, thanks!.  Hopefully I can run this myself from Jenkins?

Comment (yangchiu) [reply e2e/libs/workload/workload.py:27]: Yes, we can run customized e2e tests by building images by yourself now, but since e2e test has been refactored by https://github.com/longhorn/longhorn-tests/pull/1604, it is possible that you need to make extra changes to make things work.

**Comment by yangchiu** [comment]
I've created a [refined PR](https://github.com/longhorn/longhorn-tests/pull/1904), so close this one in favor of it.

**Pull request closed without merging**
